catalog/registry/catalog-registry-identificationplugin/src/main/java/ddf/catalog/registry/identification/IdentificationPlugin.java

```diff
@@ -76,21 +99,50 @@ public DeleteRequest process(DeleteRequest input)
         return input;
     }
 
+    @Override
+    public CreateResponse process(CreateResponse input) throws PluginExecutionException {
+        if (Requests.isLocal(input.getRequest())) {
+            for (Metacard metacard : input.getCreatedMetacards()) {
+                if (metacard.getTags()
+                        .contains(RegistryConstants.REGISTRY_TAG)) {
+                    registryIds.add(getRegistryId(metacard));
+                }
+            }
+        }
+        return input;
+    }
+
+    @Override
+    public UpdateResponse process(UpdateResponse input) throws PluginExecutionException {
+        return input;
+    }
+
+    @Override
+    public DeleteResponse process(DeleteResponse input) throws PluginExecutionException {
+        if (Requests.isLocal(input.getRequest())) {
+            for (Metacard metacard : input.getDeletedMetacards()) {
+                if (metacard.getTags()
+                        .contains(RegistryConstants.REGISTRY_TAG)) {
+                    registryIds.remove(getRegistryId(metacard));
```

Comment:
It gets its initial list on startup from the catalog via the FederationAdminService and then keeps track of them as they go through the plugin.